src/main/msc/emfat_file.c

```diff
@@ -291,44 +294,102 @@ static void emfat_add_log(emfat_entry_t *entry, int number, uint32_t offset, uin
     entry->curr_size = size;
     entry->max_size = entry->curr_size;
     entry->readcb = bblog_read_proc;
-    emfat_set_entry_cma(entry);
+    // Set file modification/access times to be the same, either the default date or that from the RTC
+    entry->cma_time[1] = cmaTime;
+    entry->cma_time[2] = cmaTime;
 }
 
 static int emfat_find_log(emfat_entry_t *entry, int maxCount)
 {
     int limit = flashfsIdentifyStartOfFreeSpace();
     int lastOffset = 0;
     int currOffset = 0;
+    int buffOffset;
+    int hdrOffset;
     int fileNumber = 0;
-    uint8_t buffer[18];
+    uint8_t buffer[HDR_BUF_SIZE];
     int logCount = 0;
+    char *timeHeader = "H Log start datetime:";
+    int timeHeaderMatched = 0;
+    int lenTimeHeader = strlen(timeHeader);
 
     for ( ; currOffset < limit ; currOffset += 2048) { // XXX 2048 = FREE_BLOCK_SIZE in io/flashfs.c
 
-        flashfsReadAbs(currOffset, buffer, 18);
+        flashfsReadAbs(currOffset, buffer, HDR_BUF_SIZE);
 
         if (strncmp((char *)buffer, "H Product:Blackbox", 18)) {
```

Comment:
Use `HDR_BUF_SIZE` instead of 18.

Reply:
No, this is the length of the string to be matched, not the buffer size. I could make the buffer a minimum of 18 bytes long, but thought is would be a little more efficient to make it a little longer. I've changed the code to use the value of lenLogHeader here.